src/cdk/constructs/cicd/index.ts

```diff
@@ -53,7 +53,8 @@ export class Cicd extends Stack {
     const sourceOutput = new Artifact()
 
     const encryptionKey = new Key(this, 'key', {
-      enableKeyRotation: true
+      enableKeyRotation: true,
+      removalPolicy: RemovalPolicy.DESTROY
```

Comment:
Only thought here is that there could sometimes be situations where preserving a key may be desired.  It's maybe not an item to address right now but could be marked as a TODO.